src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs

```diff
@@ -144,13 +151,23 @@ await functionMetadata
                 await FileUtility.WriteAsync(dataFilePath, functionMetadata.TestData);
             }
 
+            var updatedFunctionMetadata = (await _hostFunctionMetadataProvider.GetFunctionMetadataAsync(_languageWorkerOptions.CurrentValue.WorkerConfigs, configChanged))
```

Comment:
On the other path [here](https://github.com/Azure/azure-functions-host/blob/ba3e9ef990a9a5ebacaf47bbc3a71ed5f661e8ea/src/WebJobs.Script/Host/FunctionMetadataProvider.cs#L60) where this provider is used, we set FunctionErrors. We'll be bypassing that here, but maybe that's OK? Assuming that if the host restarts after this call, it'll be reindexing anyways and set the errors if any at that time.

You'll also be bypassing the AllowList that is normally applied [here](https://github.com/Azure/azure-functions-host/blob/ba3e9ef990a9a5ebacaf47bbc3a71ed5f661e8ea/src/WebJobs.Script/Host/FunctionMetadataManager.cs#L97)